src/main/java/com/squareup/kotlinpoet/TypeSpec.kt

```diff
@@ -95,7 +95,14 @@ class TypeSpec private constructor(builder: TypeSpec.Builder) {
 
         codeWriter.emitCode("object")
         val supertype = if (superclass != ANY) {
-          listOf(CodeBlock.of(" %T(%L)", superclass, anonymousTypeArguments))
+          var superclassConstructorParameters = superclassConstructorParameters
+          if (anonymousTypeArguments.isNotEmpty()) {
+            superclassConstructorParameters = superclassConstructorParameters
+                .toMutableList().apply {
+              add(anonymousTypeArguments)
```

Comment:
made this append, but I'd also like to get rid of `anonymousTypeArguments` completely, I think?